### PR TITLE
std.windows: use atomic rename, if possible

### PR DESCRIPTION
**Pull request by matu3ba**
Mitigates #14978.

### CODE REVIEW

**Comment by matu3ba** [comment]
As far as I understand, there are no tests currently. If wanted, I can add some to show that the posix behavior is what it claims.

File: lib/std/os/windows.zig [hunk 1]
```diff
@@ -2397,12 +2397,33 @@ pub const FILE_NAME_INFORMATION = extern struct {
...
+pub const FILE_RENAME_SUPPRESS_STORAGE_RESERVE_INHERITANCE = 0x00000008;
+pub const FILE_RENAME_NO_INCREASE_AVAILABLE_SPACE = 0x00000010;
+pub const FILE_RENAME_NO_DECREASE_AVAILABLE_SPACE = 0x00000020;
+pub const FILE_RENAME_PRESERVE_AVAILABLE_SPACE = 0x00000030;
+pub const FILE_RENAME_IGNORE_READONLY_ATTRIBUTE = 0x00000040;
+pub const FILE_RENAME_FORCE_RESIZE_TARGET_SR = 0x00000080;
+pub const FILE_RENAME_FORCE_RESIZE_SOURCE_SR = 0x00000100;
+pub const FILE_RENAME_FORCE_RESIZE_SR = 0x00000180;
+
+pub const FILE_RENAME_INFORMATION = if (builtin.target.os.version_range.windows.min.isAtLeast(.win10_rs1))
```
Comment (matu3ba) [inline lib/std/os/windows.zig:2413]: zig1 doesn't like these. Should I
- 1. update zig1.wasm
- 2. create FILE_RENAME_INFORMATION_EXT ?

MS docs provide this version-dependent definitition https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/ns-ntifs-_file_rename_information.
I'll try one more time to customize the type only. Maybe that works.

Comment (matu3ba) [reply lib/std/os/windows.zig:2413]: Note, that this does not contain the conditional macro lib/libc/include/any-windows-any/ddk/ntifs.h
```c
typedef struct _FILE_RENAME_INFORMATION {
  BOOLEAN ReplaceIfExists;
  HANDLE RootDirectory;
  ULONG FileNameLength;
  WCHAR FileName[1];
} FILE_RENAME_INFORMATION, *PFILE_RENAME_INFORMATION;
```

Comment (matu3ba) [reply lib/std/os/windows.zig]: Upsteam at https://git.code.sf.net/p/mingw-w64/mingw-w64 the info is missing, which is surprising. Maybe under another name stored or something.

Comment (Biswa96) [reply lib/std/os/windows.zig]: That ntifs.h is for creating kernel mode drivers. So, it can not be added in headers required for user mode applications.

**Comment by andrewrk** [comment]
Closing abandoned pull request. Please open a new one if you wish to continue this work.

**Comment by matu3ba** [comment]
Found it with in mingw64 upstream via `rg '.*FILE_RENAME.*'`: mingw-w64-headers/include/winbase.h:3122
```h
#if _WIN32_WINNT >= _WIN32_WINNT_WIN10_RS1
#define FILE_RENAME_FLAG_REPLACE_IF_EXISTS 0x00000001
#define FILE_RENAME_FLAG_POSIX_SEMANTICS 0x00000002
#endif

#if _WIN32_WINNT >= _WIN32_WINNT_WIN10_RS3
#define FILE_RENAME_FLAG_SUPPRESS_PIN_STATE_INHERITANCE 0x00000004
#endif

  typedef struct _FILE_RENAME_INFO {
#if _WIN32_WINNT >= _WIN32_WINNT_WIN10_RS1
    __C89_NAMELESS union {
        BOOLEAN ReplaceIfExists;
        DWORD Flags;
    };
#else
    BOOLEAN ReplaceIfExists;
#endif
    HANDLE RootDirectory;
    DWORD FileNameLength;
    WCHAR FileName[1];
  } FILE_RENAME_INFO,*PFILE_RENAME_INFO;
```
I'll probably make a PR next weekend unless somebody else is faster.

**Pull request closed without merging**